File: source/arduino/BoardSimulator/FlameSensor.cpp
#include "FlameSensor.h"

FlameSensor::FlameSensor(uint8_t analogPin, uint8_t digitalPin)
    : _analogPin(analogPin), _digitalPin(digitalPin) {}

void FlameSensor::begin() {
    pinMode(_digitalPin, INPUT);
}

int FlameSensor::readAnalog() {
    return analogRead(_analogPin);
}

bool FlameSensor::readDigital() {
    return digitalRead(_digitalPin) == HIGH;
}

File: source/rpi/BoardSimulator/flame_sensor.py
# flame_sensor.py
# Flame sensor simulation for Raspberry Pi

class FlameSensor:
    def __init__(self, analog_pin, digital_pin):
        self.analog_pin = analog_pin
        self.digital_pin = digital_pin

    def begin(self):
        print(f"Flame sensor initialized on analog pin {self.analog_pin}, digital pin {self.digital_pin}")

    def read_analog(self):
        # Simulate analog value
        value = 123  # Replace with actual reading logic if needed
        print(f"FlameSensor Analog: {value}")
        return value

    def read_digital(self):
        # Simulate digital value
        value = True  # Replace with actual reading logic if needed
        print(f"FlameSensor Digital: {'HIGH' if value else 'LOW'}")
        return value
